spec/index/index_spec.rb

```diff
@@ -70,6 +70,40 @@
     end
   end
 
+  context '#sort' do
+    let(:idx1) { Daru::Index.new ['speaker', 'mic', 'guitar', 'amp'] }
+    let(:idx2) { Daru::Index.new [:one, 'one', 1, 2, :two] }
+    let(:idx3) { Daru::Index.new [100, 99, 101, 1, 2] }
+    let(:idx4) { Daru::Index.new [11, Float::NAN, nil] }
+    let(:idx1_asc) { Daru::Index.new ['amp', 'guitar', 'mic', 'speaker'] }
+    let(:idx1_desc) { Daru::Index.new ['speaker', 'mic', 'guitar', 'amp'] }
+    let(:idx3_asc) { Daru::Index.new [1, 2, 99, 100, 101] }
+    let(:idx3_desc) { Daru::Index.new [101, 100, 99, 2, 1] }
+    let(:idx3_asc_new_order) { [3, 4, 1, 0, 2] }
+    let(:idx3_desc_new_order) { [2, 0, 1, 4, 3] }
+    let(:error_msg) { 'All elements are not of the same type/class.' }
+
+    it 'default order is ascending' do
+      expect(idx1.sort()).to eq(idx1_asc)
+      expect(idx3.sort()).to eq(idx3_asc)
+    end
+
+    it 'set ascending to false' do
+      expect(idx1.sort(ascending: false)).to eq(idx1_desc)
+      expect(idx3.sort(ascending: false)).to eq(idx3_desc)
+    end
+
+    it 'To get the new order, set get_new_order to true' do
+      expect(idx3.sort(get_new_order: true)).to eq [idx3_asc, idx3_asc_new_order]
+      expect(idx3.sort(ascending: false, get_new_order: true)).to eq [idx3_desc, idx3_desc_new_order]
+    end
+
+    it "raises ArgumentError if all the values are not of the same class" do
```

Comment:
`expect { idx.sort }.to raise_error` (curly braces). The idea is you pass some code to `expect`, and what you expect it to do (raise error). And then it is checked (code is run, and error is caught). With syntax `expect(idx2.sort)` what is going on is:
1. sort is called
2. expect is called with result of `sort` call (or, rather it is not, because error is already raised).

BTW, are you aware you can test it locally, just calling `rspec` command?